### DISCUSSION TITLE
Allow access to 'blockData' in hooks/conditions

### DISCUSSION THREAD

**Discussion by christian-reichart** [payloadcms/payload#1665] [Feature Requests & Ideas]
I now often times had the problem that I wanted to conditionally show a field inside a block depending on another field in the block. Whenever the specific field is placed inside a group/array/..., I can't access the field to base the condition on with siblingData. Using the whole document is also not practical, as I only need  to care about the block scope.

Is this something that could be added?

**Comment by DanRibbens** [comment]
I have experienced this and worked around the problem by changing my data model just to simplify conditional logic. 

I am curious what would make writing complex conditionals easier?

Perhaps some helper functions that take the full data object and help with traversing documents a new passed in path property?

**Reply by christian-reichart** [reply]
So for me the only way I use the params provided by the condition function is by using `siblingData`. I've never had a use case where I use the document data because I mainly work with blocks.

Here is a simple example of how the problem would be solved for me with `blockData`, but that's maybe too specific:

```typescript
const Demo: Block = {
  slug: 'demo',
  fields: [
    {
      name: 'variant',
      label: 'Variant',
      type: 'select',
      options: [
        {
          value: 'A',
          label: 'A',
        },
        {
          value: 'B',
          label: 'B',
        },
      ],
    },
    {
      name: 'demoGroup',
      type: 'group',
      fields: [
        {
          name: 'conditionField',
          type: 'text',
          admin: {
            condition: (data, blockData, siblingData) => {
              return blockData.variant === 'A'
            }
          }
        }
      ]
    }
  ],
}
```

I didn't do a deep dive yet to see if this is easily possible to provide. I'm also not sure if this could be generalized so it's a helpful functionality for collections too. Something like `parentData` or a helper function to "go up" a certain number of parents could maybe solve this also.

`const parent = getFieldParent({ level: 2 })`

**Comment by brishan3** [comment]
Bumping this issue as my team and I were discussing this problem recently. We are currently rebuilding our company website using Payload CMS and NextJS 13.

I definitely agree with @christian-reichart, there should truly be a low friction way to access fields held within a Block when using conditional logic on fields that are nested within groups or arrays.
